src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/IDependenciesChanges.cs

```diff
@@ -31,5 +35,25 @@ public void Deconstruct(out string providerType, out string dependencyId)
             providerType = ProviderType;
             dependencyId = DependencyId;
         }
+
+        public bool Equals(RemovedDependencyIdentity other)
+        {
+            return string.Equals(ProviderType, other.ProviderType, StringComparisons.DependencyProviderTypes) && 
+                   string.Equals(DependencyId, other.DependencyId, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override bool Equals(object obj) => obj is RemovedDependencyIdentity other && Equals(other);
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                return (StringComparer.OrdinalIgnoreCase.GetHashCode(ProviderType) * 397) ^
+                        StringComparer.OrdinalIgnoreCase.GetHashCode(DependencyId);
```

Comment:
Late, and doesn't apply as well when you need `OrdinalIgnoreCase`, but a handy trick to make the compiler generate this (and .Equals), is to use tuples under the hood:

```C#
    return (ProviderType, DependencyId).GetHashCode();
```

and 

```C#
    return (ProviderType, DependencyId) == (other.ProviderType, other.DependencyId);
```

Reply:
We can use this now we are on C# 7.3.

https://docs.microsoft.com/en-us/dotnet/csharp/tuples#equality-and-tuples